src/clj/rems/api/catalogue_items.clj

```diff
@@ -69,16 +69,16 @@
       :return CreateCatalogueItemResponse
       (ok (catalogue/create-catalogue-item! command)))
 
-    (PUT "/update" []
-      :summary "Update catalogue item"
+    (POST "/edit" []
+      :summary "Edit a catalogue item"
       :roles #{:owner}
-      :body [command UpdateStateCommand]
+      :body [command EditCatalogueItemCommand]
       :return SuccessResponse
-      (ok (catalogue/update-catalogue-item! command)))
+      (ok (catalogue/edit-catalogue-item! command)))
 
-    (POST "/create-localization" []
-      :summary "Create a new catalogue item localization"
+    (PUT "/update" []
```

Comment:
I can't see any meaningful difference between `/edit` and `/update`. How about just adding new fields to UpdateStateCommand?

Reply:
Hmm. So, this naming is analogous to how similar operations are named in forms. There the distinction is meaningful because updating form [state], i.e., changing archived/disabled status, is always doable, whereas editing form can only be done if the form is not in use. Also, editing form shares schema with creating a new form, so it made sense there to keep the operations distinct.

There's also UpdateStateCommand defined in Schemas, which consists of fields id, enabled, and archived, and which is in use for catalogue items, forms, licenses, and resources.

However, workflow has UpdateWorkflowCommand, which includes fields for editing the actual workflow (i.e., handlers and title).

Based on this, I'm not quite sure which route to take. My feeling is that I'd like to keep things modular and extract the common operations (archiving/disabling) to their own API call to have a more uniform API over different kinds of "elements" (resources, workflows, etc.) However, the naming is arguably confusing -- update should probably be update-state or similar.

Reply:
Hmmm ok you're right. Maybe the operations need to be separate, but I'd like better names. We can go with update for now but could you create a technical debt ticket about renaming update to update-state (or whatever)?

Reply:
Done. https://github.com/CSCfi/rems/issues/1548